services/management/external.go

```diff
@@ -82,6 +83,20 @@ func (e ExternalService) AddExternal(ctx context.Context, req *managementpb.AddE
 		}
 		res.Service = invService.(*inventorypb.ExternalService)
 
+		agentIDs, err := models.FindPMMAgentsRunningOnNode(tx.Querier, req.RunsOnNodeId)
+		if err != nil {
+			return errors.Wrapf(err, "cannot find pmm_agent for external exporter with push_metrics")
+		}
+		if len(agentIDs) > 1 {
+			return errors.Errorf("cannot find exact match for pmm_agent for external exporter,"+
```

Comment:
if we have more than one let's set pull as well.

Reply:
if user didn't set metrics-mode push explicitly

Reply:
Ok, behavior changed.

Case: AUTO && (agents count > 1 or err != nil) => PULL
Case: AUTO && count == 1 && err == nil => check for supported mode

Othervise metrcis-mode will be without change.